### PR TITLE
MM-39711 Gekidou Settings - Display Main Screen 

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
Similar to https://github.com/mattermost/mattermost-mobile/pull/6285 , with this PR, we bring in the main screen for settings/display.

This PR is to be merged only after https://github.com/mattermost/mattermost-mobile/pull/6285 is merged into gekidou as #6285  brought some refactoring.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39711

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/screens/settings/display/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import {getAllowedThemes, observeConfigBooleanValue} from '@queries/servers/system';
+import {WithDatabaseArgs} from '@typings/database/database';
+
+import DisplaySettings from './display';
+
+const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
+    const isTimezoneEnabled = observeConfigBooleanValue(database, 'ExperimentalTimezone');
+
+    const allowsThemeSwitching = observeConfigBooleanValue(database, 'EnableThemeSelection');
+    const allowedThemes = from$(getAllowedThemes(database));
```
Comment (larkox) [inline app/screens/settings/display/index.tsx:18]: Why do we observe a get, instead of creating an observe function directly?

**Review by larkox — APPROVED** [review]
LGTM, just one potential change.

File: app/queries/servers/system.ts [hunk 2]
```diff
@@ -395,3 +396,19 @@ export const observeOnlyUnreads = (database: Database) => {
         switchMap((model) => of$(model.value as boolean)),
     );
 };
+
+export const getAllowedThemes = async (database: Database): Promise<string[]> => {
```
Comment (enahum) [inline app/queries/servers/system.ts:400]: you should make this an observable

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/settings/display/index.tsx (hunk 1 above)
Comment (avinashlng1080) [reply app/screens/settings/display/index.tsx]: corrected

**Comment by enahum** [comment]
@avinashlng1080 conflicts

**Review by enahum — CHANGES_REQUESTED** [review]
missing translation strings

**Review by enahum — APPROVED** [review]

**Pull request merged**